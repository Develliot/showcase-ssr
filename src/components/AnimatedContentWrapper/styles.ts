import styled from 'styled-components';
import { ContentWrapper } from 'src/components/Layout';

export const ContentWrapperStyled = styled(ContentWrapper)`
    transition: opacity ease 0.5s, transform ease 0.4s;
    opacity: ${({ visible }: { visible: boolean }) => (visible ? 1 : 0)};
    transform: ${({ visible }: { visible: boolean }) =>
        visible ? `scaleY(1)` : `scaleY(0.85)`};
`;
