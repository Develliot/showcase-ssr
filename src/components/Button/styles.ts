import styled from 'styled-components';

export const ButtonStyled = styled.button`
    font-size: ${({
        theme: {
            fontSizes: { medium },
        },
    }) => medium};
    padding: ${({
        theme: {
            spacing: { tiny, medium },
        },
    }) => `${tiny} ${medium}`};
    font-weight: 500;
    text-transform: uppercase;
    transition: color 0.3s ease, background-color 0.3s ease;
    background-color: ${({
        color,
        theme: {
            colors,
            colors: { blue },
        },
    }) => (color ? colors[`${color}`] : blue)};
    border-width: 0.12rem;
    border-style: solid;
    border-radius: 1.5rem;
    border-color: ${({
        color,
        theme: {
            colors,
            colors: { blue },
        },
    }) => (color ? colors[`${color}`] : blue)};
    color: ${({
        color,
        theme: {
            colors: { white, darkGrey },
        },
    }) => {
        switch (color) {
            case 'white':
                return darkGrey;
            case 'yellow':
                return darkGrey;
            case 'blue':
                return white;
            case 'pink':
                return white;
            case 'darkGrey':
                return white;
            case undefined:
                return white;
            default:
                return white;
        }
    }};

    &:hover,
    &:focus {
        background-color: ${({
            color,
            theme: {
                colors: { white },
            },
        }) => {
            switch (color) {
                case 'white':
                    return `rgba(0, 0, 0, 0.4)`;
                case 'yellow':
                    return `rgba(0, 0, 0, 0.4)`;
                case 'blue':
                    return white;
                case 'pink':
                    return white;
                case 'darkGrey':
                    return white;
                case undefined:
                    return white;
                default:
                    return white;
            }
        }};
        color: ${({
            color,
            theme: {
                colors,
                colors: { blue },
            },
        }) => (color ? colors[`${color}`] : blue)};
    }
`;
