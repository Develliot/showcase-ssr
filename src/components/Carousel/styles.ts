import styled from 'styled-components';

export const CarouselWrapper = styled.div`
    position: relative;
    width: 100%;
    overflow-y: hidden;
`;

export const CarouselFadeOutLeft = styled.div`
    pointer-events: none;
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    transition: opacity 0.3s linear;
    opacity: ${({ visible }: { visible: boolean }) => (visible ? 1 : 0)};
    background: ${({
        theme: {
            colors: { blue },
        },
    }) =>
        `linear-gradient(to right, ${blue} , rgba(0,0,0,0) 10%, rgba(0,0,0,0) 90%)`};
`;

export const CarouselFadeOutRight = styled(CarouselFadeOutLeft)`
    background: ${({
        theme: {
            colors: { blue },
        },
    }) =>
        `linear-gradient(to left, ${blue} , rgba(0,0,0,0) 10%, rgba(0,0,0,0) 90%)`};
`;

export const CarouselScrollArea = styled.div`
    overflow-x: scroll;
    -webkit-overflow-scrolling: touch;
    scroll-snap-type: x mandatory;
    scroll-behavior: smooth;
    white-space: nowrap;
    /* to hides */
    padding-bottom: 20px;
    margin-bottom: -20px;
    height: 100%;
`;

export const CarouselCard = styled.div`
    scroll-snap-align: center;
    display: inline-block;
    width: 80%;
`;

export const CarouselCardInner = styled.div`
    transition: transform 0.2s linear;
    transform: ${({ isSelected }: { isSelected: boolean }) =>
        isSelected ? 'scale(1)' : 'scale(0.85)'};
`;

export const LeftButtonContainer = styled.div`
    position: absolute;
    display: flex;
    left: 0;
    top: 0;
    bottom: 0;
`;

export const RightButtonContainer = styled(LeftButtonContainer)`
    left: auto;
    right: 0;
`;

export const CarouselButton = styled.button`
    background-color: rgba(0, 0, 0, 0);
    border: none;
    color: ${({
        theme: {
            colors: { white },
        },
    }) => white};
    font-size: ${({
        theme: {
            fontSizes: { large },
        },
    }) => large};
    font-weight: 600;
`;

export const DotContainer = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: center;
`;

export const Dot = styled.button`
    border: none;
    padding: 0;
    margin: ${({
        theme: {
            spacing: { tiny },
        },
    }) => tiny};

    width: 16px;
    height: 16px;
    border-radius: 50%;
    color: ${({
        theme: {
            colors: { white },
        },
    }) => white};
    transition: opacity 0.3s ease, transform 0.3s ease;
    opacity: ${({ selected }: { selected: boolean }) => (selected ? 1 : 0.6)};

    &:hover {
        transform: scale(1.5);
    }
`;
