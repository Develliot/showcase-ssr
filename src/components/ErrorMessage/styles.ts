import styled from 'styled-components';

export const ErrorWrapper = styled.div`
    background-color: ${({
        theme: {
            colors: { white },
        },
    }) => white};
    padding: ${({
        theme: {
            spacing: { medium },
        },
    }) => medium};
    border-radius: 1rem;
`;
