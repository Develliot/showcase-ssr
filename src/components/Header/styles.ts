import styled from 'styled-components';

export const HeaderWrapper = styled.header`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    margin: 0 auto;
    width: 100vw;
    max-width: ${({
        theme: {
            breakpoints: { large },
        },
    }) => large};
    transition: margin ease 0.2s, opacity ease 0.3s;
    margin-top: ${({ visible }: { visible: boolean }) =>
        visible ? `0` : `-100px`};
    opacity: ${({ visible }: { visible: boolean }) => (visible ? `1` : `0`)};
    z-index: 10;
`;

export const HeaderInner = styled.div`
    margin: 0 auto;
    background-color: ${({
        theme: {
            colors: { darkGrey },
        },
    }) => darkGrey};
    padding: ${({
        theme: {
            spacing: { small },
            globalPadding,
        },
    }) => `${small} ${globalPadding}`};
`;

export const Links = styled.ul`
    padding: 0;
    max-width: ${({
        theme: {
            breakpoints: { large },
        },
    }) => large};

    display: flex;
    flex-direction: row;
    list-style-type: none;
    justify-content: flex-start;
`;

export const LogoLink = styled.a`
    text-decoration: none;
`;
