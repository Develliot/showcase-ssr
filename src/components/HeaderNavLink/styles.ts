import styled from 'styled-components';
import { Paragraph } from 'src/components/Typography';

export const HeaderLinkItem = styled.li`
    padding: 0;
    margin: ${({
        theme: {
            spacing: { small },
        },
    }) => `0 ${small} 0 0`};
`;

export const HoverInidcator = styled.div`
    height: 3px;
    background-color: ${({
        theme: {
            colors: { yellow },
        },
    }) => yellow};
    margin-top: ${({
        theme: {
            spacing: { tiny },
        },
    }) => tiny};
    width: 100%;
    max-width: 0;
    transition: max-width ease 0.3s;

    max-width: ${({ selected }: { selected: boolean }) =>
        selected ? `150px` : 0};
`;

export const RowCenter = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: center;
`;

export const NavLinkStyled = styled.a`
    text-decoration: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
    ${Paragraph} {
        transition: color ease 0.3s;
        font-weight: 600;
    }
    cursor: pointer;

    &:hover,
    &:focus {
        outline: none;

        ${Paragraph} {
            color: ${({
                theme: {
                    colors: { yellow },
                },
            }) => yellow};
        }
    }
`;
