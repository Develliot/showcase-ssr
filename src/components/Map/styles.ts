import styled from 'styled-components';

export const MapWrapper = styled.div`
    width: 100%;
    height: 300px;
    background-color: ${({
        theme: {
            colors: { white },
        },
    }) => white};
`;

export const MarkerWrapper = styled.div`
    height: 30px;
    width: 30px;
    border-radius: 50%;
    overflow: hidden;
    border: ${({
        theme: {
            colors: { darkGrey },
        },
    }) => `solid 1px ${darkGrey}`};
`;

export const MarkerImage = styled.img`
    height: 100%;
    width: 100%;
`;
