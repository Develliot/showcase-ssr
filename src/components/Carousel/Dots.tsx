import React, { FunctionComponent } from 'react';
import { DotContainer, Dot } from './styles';

type DotProps = {
    count: number;
    currentPosition: number;
    scrollToPosition: (to: number) => void;
};
export const Dots: FunctionComponent<DotProps> = ({
    count,
    currentPosition,
    scrollToPosition,
}) => {
    return (
        <DotContainer>
            {[...Array(count)].map((item, index) => {
                const isSelected = index === currentPosition;
                const scrollTo = () => {
                    scrollToPosition(index);
                };
                return (
                    <Dot key={index} selected={isSelected} onClick={scrollTo} />
                );
            })}
        </DotContainer>
    );
};

export default Dots;
