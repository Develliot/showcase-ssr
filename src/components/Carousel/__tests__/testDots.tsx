import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Dots } from '../Dots';
import { Dot } from '../styles';

it('renders the correct selected dot', () => {
    const wrapper: ShallowWrapper = shallow(
        <Dots count={4} currentPosition={3} scrollToPosition={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
});

it('fires the correct scroll to function after a dot is clicked', () => {
    const mockScrollToPostion = jest.fn();

    const wrapper: ShallowWrapper = shallow(
        <Dots
            count={4}
            currentPosition={3}
            scrollToPosition={mockScrollToPostion}
        />
    );
    wrapper
        .find(Dot)
        .first()
        .simulate('click');
    expect(mockScrollToPostion).toHaveBeenCalledWith(0);
});
