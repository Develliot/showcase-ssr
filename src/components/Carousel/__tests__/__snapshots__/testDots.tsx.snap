// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct selected dot 1`] = `
<styles__DotContainer>
  <styles__Dot
    key="0"
    onClick={[Function]}
    selected={false}
  />
  <styles__Dot
    key="1"
    onClick={[Function]}
    selected={false}
  />
  <styles__Dot
    key="2"
    onClick={[Function]}
    selected={false}
  />
  <styles__Dot
    key="3"
    onClick={[Function]}
    selected={true}
  />
</styles__DotContainer>
`;
